main.go

```diff
@@ -280,49 +278,41 @@ func (s *Session) firstNav(ctx context.Context) error {
 	return nil
 }
 
-// navToEnd waits for the page to be ready to receive scroll key events, by
-// trying to select an item with the right arrow key, and then scrolls down to the
-// end of the page, i.e. to the oldest items.
+// navToEnd selects the last item in the page
+//  by repeatedly advances the selected item with
+//  - kb.ArrowRight (which causes an initial selection, and/or advances it by one)
+//  - kb.End which scrolls to the end of the page, and advances the selected item.
+// Note timing is important, because when the kb.End causes significant scrolling,
+//  the active element become undefined for a certain time, in that case, we get an error (ignore), sleep, and retry.
+// The termnation criteria is that the selected item (document.activeElement.href) is stable for >2 iterations
 func navToEnd(ctx context.Context) error {
-	// wait for page to be loaded, i.e. that we can make an element active by using
-	// the right arrow key.
+	var prev, active string
+	lastRepeated := 0
 	for {
 		chromedp.KeyEvent(kb.ArrowRight).Do(ctx)
+		chromedp.KeyEvent(kb.End).Do(ctx)
 		time.Sleep(tick)
-		var ids []cdp.NodeID
-		if err := chromedp.Run(ctx,
-			chromedp.NodeIDs(`document.activeElement`, &ids, chromedp.ByJSPath)); err != nil {
-			return err
+
+		if err := chromedp.Evaluate(`document.activeElement.href`, &active).Do(ctx); err != nil {
+			time.Sleep(tick) // this extra sleep is important: after the kb.End, it sometimes takes a while for the active element to be reset
+			continue         // ignore this error: no active element, or active element has no href
 		}
-		if len(ids) > 0 {
-			if *verboseFlag {
-				log.Printf("We are ready, because element %v is selected", ids[0])
-			}
-			break
+		if active == prev {
+			lastRepeated++
+		} else {
+			lastRepeated = 0
 		}
-		time.Sleep(tick)
-	}
-
-	// try jumping to the end of the page. detect we are there and have stopped
-	// moving when two consecutive screenshots are identical.
-	var previousScr, scr []byte
-	for {
-		chromedp.KeyEvent(kb.PageDown).Do(ctx)
-		chromedp.KeyEvent(kb.End).Do(ctx)
-		chromedp.CaptureScreenshot(&scr).Do(ctx)
-		if previousScr == nil {
-			previousScr = scr
-			continue
+		if *verboseFlag {
+			log.Printf("** navToEnd:activeElt %s %d", active, lastRepeated)
 		}
-		if bytes.Equal(previousScr, scr) {
+		if lastRepeated > 2 {
```

Comment:
why is

    if lastRepeated > 1 {

not enough?

Reply:
It does happen, although very rarely, that the navigation (ArrowRight) stutters (does not advance). I discovered this after running the code hundreds of times in a burn-in test.

Reply:
hmm, then can you add a comment about it please?